contracts/crashpunks-v2-friedger.clar

```diff
@@ -0,0 +1,345 @@
+            (shares (var-get collection-royalty-shares))
+            (split u0)
+        )
+        (+ split (unwrap! (pay-royalty payer saleAmount (unwrap! (nft-get-owner? crashpunks-v2 nftIndex) ERR-PAYMENT-ADDRESS) (unwrap! (element-at shares u0) ERR-PAYMENT-ADDRESS)) ERR-PAYMENT-ADDRESS))
+        (+ split (unwrap! (pay-royalty payer saleAmount (unwrap! (element-at addresses u1) ERR-PAYMENT-ADDRESS) (unwrap! (element-at shares u1) ERR-PAYMENT-ADDRESS)) ERR-PAYMENT-ADDRESS))
+        (+ split (unwrap! (pay-royalty payer saleAmount (unwrap! (element-at addresses u2) ERR-PAYMENT-ADDRESS) (unwrap! (element-at shares u2) ERR-PAYMENT-ADDRESS)) ERR-PAYMENT-ADDRESS))
+        (+ split (unwrap! (pay-royalty payer saleAmount (unwrap! (element-at addresses u3) ERR-PAYMENT-ADDRESS) (unwrap! (element-at shares u3) ERR-PAYMENT-ADDRESS)) ERR-PAYMENT-ADDRESS))
+        (+ split (unwrap! (pay-royalty payer saleAmount (unwrap! (element-at addresses u4) ERR-PAYMENT-ADDRESS) (unwrap! (element-at shares u4) ERR-PAYMENT-ADDRESS)) ERR-PAYMENT-ADDRESS))
+        (+ split (unwrap! (pay-royalty payer saleAmount (unwrap! (element-at addresses u5) ERR-PAYMENT-ADDRESS) (unwrap! (element-at shares u5) ERR-PAYMENT-ADDRESS)) ERR-PAYMENT-ADDRESS))
+        (+ split (unwrap! (pay-royalty payer saleAmount (unwrap! (element-at addresses u6) ERR-PAYMENT-ADDRESS) (unwrap! (element-at shares u6) ERR-PAYMENT-ADDRESS)) ERR-PAYMENT-ADDRESS))
+        (+ split (unwrap! (pay-royalty payer saleAmount (unwrap! (element-at addresses u7) ERR-PAYMENT-ADDRESS) (unwrap! (element-at shares u7) ERR-PAYMENT-ADDRESS)) ERR-PAYMENT-ADDRESS))
+        (+ split (unwrap! (pay-royalty payer saleAmount (unwrap! (element-at addresses u8) ERR-PAYMENT-ADDRESS) (unwrap! (element-at shares u8) ERR-PAYMENT-ADDRESS)) ERR-PAYMENT-ADDRESS))
+        (+ split (unwrap! (pay-royalty payer saleAmount (unwrap! (element-at addresses u9) ERR-PAYMENT-ADDRESS) (unwrap! (element-at shares u9) ERR-PAYMENT-ADDRESS)) ERR-PAYMENT-ADDRESS))
+        (ok split)
+    )
+)
+
+(define-private (pay-royalty (payer principal) (saleAmount uint) (payee principal) (share uint))
+    (let ((split (/ (* saleAmount share) percentage-with-twodp)))
+        ;; ignore royalty payment if its to the buyer / contract-caller.
+        (if (and 
+                (> share u0)
+                (not (is-eq contract-caller payee)) 
+                (try! (stx-transfer? split payer payee))
+            )
+            (ok split)
+            (ok u0)
+        )
+    )
+)
+
+(define-private (upgrade-v1-to-v2-helper (nftIndex uint) (initial-value bool))
+    (unwrap-panic (upgrade-v1-to-v2 nftIndex))
+)
+
+(define-private (mint-token-helper (entry uint) (initial-value bool))
+    (unwrap-panic (mint-token))
+)
+
+(define-private (set-mint-pass-helper (entry {account: principal, limit: uint}) (initial-value bool))
+    (unwrap-panic (set-mint-pass (get account entry) (get limit entry)))
+)
+
+;; TODO: add all whitelists
```

Comment:
Store these in a separate contract, to have a light weight contract after minting.

Reply:
we most likely wont have the white lists ready when we deploy this contract anyway tbh